src/nomad_simulations/schema_packages/general.py

```diff
@@ -262,19 +266,84 @@ def set_composition_formula(system_parent: ModelSystem) -> None:
         for child in system_parent.sub_systems:
             Simulation.set_composition_formula(system_parent=child)
 
+    @log
+    def _validate_and_set_representative_system(self) -> None:
+        """
+        Ensure exactly one representative `ModelSystem` exists in `self.model_system`,
+        and update `self.representative_system_index` to point to it.
+
+        Behavior
+        --------
+        - **Zero representatives:** Mark the *last* item in `self.model_system` as
+        representative and set `self.representative_system_index = -1`
+        (Python’s negative index for the last element).
+        - **Multiple representatives:** Log a warning, demote all, then keep the
+        *last* representative encountered. Store its (non-negative) index in
+        `self.representative_system_index`.
+        - **Exactly one representative:** Leave flags as is and set
+        `self.representative_system_index` to that item’s index.
+
+        Side Effects
+        ------------
+        - Mutates `is_representative` on elements of `self.model_system`.
+        - Sets `self.representative_system_index`.
+        - Emits a warning if multiple representatives are found.
+
+        Requirements / Notes
+        --------------------
+        - Assumes `self.model_system` is **non-empty**. If it can be empty in your
+        context, guard against `IndexError` before calling or extend this method
+        to handle the empty case explicitly.
+        - Runs in O(n) over the number of systems.
+
+        Returns
+        -------
+        None
+        """
+        logger = self._validate_and_set_representative_system.__annotations__['logger']
+
+        # indices of representative systems
+        rep_idx = [i for i, ms in enumerate(self.model_system) if ms.is_representative]
+
+        if len(rep_idx) == 0:
+            self.model_system[-1].is_representative = True
+            self.representative_system_index = -1
```

Comment:
This will break when `self.model_system` is `None`. In case of an empty list, it will point to a non-existing position. I think both can be handled without relying on `@log` here.

Reply:
Ow sorry, the check is handled below in the normalizer. Given the completeness of this method, may consider moving it up? I leave it up to you.

Reply:
I think it needs to be in the normalizer or else the rest of the normalization will try to run after the error, and then you will have explicit run time errors, right?

Reply:
Very much possible, hence "I leave it up to you".
Maybe just add to the description that the error handling is done up.